src/Autofac.Extras.MvvmCross/AutofacMvxIocProvider.cs

```diff
@@ -482,7 +538,17 @@ public object Resolve(Type type)
             if (type == null)
                 throw new ArgumentNullException("type");
 
-            return _container.Resolve(type);
+            try
+            {
+                return _container.Resolve(type);
+            }
+            catch (DependencyResolutionException ex)
+            {
+                // throw MvxIoCResolveException
+                if (ex.InnerException is MvxIoCResolveException)
```

Comment:
That is a good question.
Principally I am with you as I do not want to loose information.
On the other hand there is external code which does not know anything about the implementation of the plugin and can therefore only expect exceptions of type MvxIoCResolveException.
I will re-work that to wrap the DependencyResolutionException in an MvxIoCException.
That way we'll have the best of both worlds